itests/testdata/transfer_tx_testdata.go

```diff
@@ -49,38 +71,387 @@ type CommonTransferData struct {
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "is more than 7200000ms in the past relative to previous block timestamp",
+			}),
+		"Timestamp more than 5400000ms in the future relative to previous block timestamp": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&assetId,
+			nil,
+			utl.MinTxFeeWaves,
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs()+54160000,
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "is more than 5400000ms in the future relative to block timestamp",
+			}),
+		"Invalid asset ID": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&invalidAssetId,
+			nil,
+			utl.MinTxFeeWaves,
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs(),
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "Attempt to transfer unavailable funds: Transaction application leads to negative asset",
+			}),
+		"Transfer token when there are not enough funds on the sender balance": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&assetId,
+			nil,
+			uint64(100000000+utl.GetAvailableBalanceInWavesGo(suite, utl.GetAccount(suite, utl.DefaultSenderNotMiner).Address)),
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs(),
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "Transaction application leads to negative waves balance",
+			}),
+	}
+	return t
+}
+
+func GetTransferChainIDNegativeData(suite *f.BaseSuite, assetId crypto.Digest) map[string]TransferTestData[TransferExpectedValuesNegative] {
+	assetAmount := utl.GetAssetBalanceGo(suite, utl.GetAccount(suite, utl.DefaultSenderNotMiner).Address, assetId)
+	var t = map[string]TransferTestData[TransferExpectedValuesNegative]{
+		"Invalid chainID (value=0)": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
```

Comment:
For transfer transactions where `version <= 2` `ChainID` is not included in binary representation of transaction because it is included in `Recipient` address. So if you want to check transaction not only where `version >= 3` you need to change `ChainID` of the recipient address.

Reply:
New test was added for this case.